src/sandstorm/run-bundle.c++

```diff
@@ -1270,6 +1313,23 @@ private:
       }
     }
 
+    // Now process the PORT setting, since the actual value in config.ports
+    // depends on if HTTPS_PORT was provided at any point in reading the
+    // config file.
+    //
+    // Outer KJ_IF_MAYBE so we only run this code if the config file contained
+    // a PORT= declaration.
+    KJ_IF_MAYBE(portValue, maybePortValue) {
+        KJ_IF_MAYBE(ports, getPorts(config.httpsPort, *portValue)) {
+            // Memory leak? Who owns this pointer? I guess it has to stay
+            // alive as long as the program does, since we don't typically
+            // delete the Config object.
+            config.ports = kj::mv(*ports);
+        } else {
+          KJ_FAIL_REQUIRE("invalid config value PORT", *portValue);
```

Comment:
I think you have two `return nullptr`s, and it seems to me they'd benefit from slightly different error messages: One is "PORT can't be empty" and the other is "PORT values must be integers". So you should have two separate KJ_FAIL_REQUIREs for the two errors; no need for a goto.

(Although I think we also determined that the first one will never happen since split() never returns an empty array.)